logic/src/commonMain/kotlin/com/wire/kalium/logic/data/connection/ConnectionRepository.kt

```diff
@@ -151,10 +151,14 @@ internal class ConnectionDataSource(
     ) {
         wrapStorageRequest {
             connections.forEach { connection ->
+                val qualifiedConversationId = idMapper.fromApiToDao(connection.qualifiedConversationId)
+                // If the conversation doesn't exists, we skip it, since the backend doesn't know this conv (probably status = CANCELLED)
+                conversationDAO.observeGetConversationByQualifiedID(qualifiedConversationId).first() ?: return
+
                 conversationDAO.updateOrInsertOneOnOneMemberWithConnectionStatus(
                     userId = idMapper.fromApiToDao(connection.qualifiedToId),
                     status = connectionStatusMapper.fromApiToDao(state = connection.status),
-                    conversationID = idMapper.fromApiToDao(connection.qualifiedConversationId)
+                    conversationID = qualifiedConversationId
                 )
             }
```

Comment:
since the whole list is handled inside one `wrapStorageRequest` any error happens inside with break the for loop and return an error without continuing to the rest or he list items., e.g. with a list of 10 connections if the first one fails then the other 9 will not be added even if they might be valid connection requests.

Reply:
Makes sense, I moved the `wrapStorageRequest` to every item call